nixos/modules/services/networking/cloudflared.nix

```
@@ -368,21 +421,28 @@ in
           RuntimeDirectory = "cloudflared-tunnel-${name}";
           RuntimeDirectoryMode = "0400";
           LoadCredential = [
-            "credentials.json:${tunnel.credentialsFile}"
-          ] ++ (lib.optional (certFile != null) "cert.pem:certFile");
-
-          ExecStart = "${cfg.package}/bin/cloudflared tunnel --config=${mkConfigFile} --no-autoupdate run";
+            (lib.optional (tunnel.credentialsFile != null) ("credentials.json:${tunnel.credentialsFile}"))
+          ] ++ (lib.optional (certFile != null) "cert.pem:${certFile}");
+
+          ExecStart =
+            if tunnel.token != null then
+              "${cfg.package}/bin/cloudflared tunnel run --token ${lib.escapeShellArg tunnel.token}"
+            else if tunnel.tokenFile != null then
+              "${cfg.package}/bin/cloudflared tunnel run --token-file ${lib.escapeShellArg tunnel.token}"
```

you might also want to use the loadcredential mechanism for the token file.

---

Could you clarify a bit? Are you talking about the .pem file?

---

the module you are modifying loads the old credentialfile via the systemd loadcredential mechanism. You might also want to do that for your newly implemented tokenfile.

---

Also there seems to be a typo in tunnel.tokenFile
```suggestion
              "${cfg.package}/bin/cloudflared tunnel run --token-file ${lib.escapeShellArg tunnel.tokenFile}"
```